092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,300 @@
+# Self Healing using Cruise Control
+
+This proposal is about enabling the self-healing properties of Cruise Control in the Strimzi operator.
+The self-healing feature of Cruise Control allows us to heal the anomalies detected in the cluster, automatically. 
+Anomalies are detected by Cruise Control using the anomaly detector manager.
+
+## Current situation
+
+During normal operations it's common for Kafka clusters to become unbalanced (through partition key skew, uneven partition placement etc.) or for problems to occur with disks or other underlying hardware which makes the cluster unhealthy.
+Currently, if we encounter any such scenario we need to fix these issues manually i.e. if there is some goal violation and the cluster is imbalanced then we might instruct Cruise Control to move the partition replicas across the brokers in order to fix the violated goal by using the `KafkaRebalance` custom resource.
+With smaller clusters, it is feasible to fix things manually. However, for larger ones it can be very time-consuming, or just not feasible, to fix all the anomalies on your own.
+
+## Motivation
+
+With the help of Cruise Control's self-healing feature we can resolve the anomalies causing unhealthy/imbalanced cluster automatically.
+The detection of anomalies is the responsibility of the anomaly detector mechanism.
+Notifiers in Cruise Control are used to send notification to user about the self-healing operations being performed in the cluster.
+Notifiers can be useful as they provide visibility to the user regarding the actions that are being performed by Cruise Control regarding self-healing for e.g. anomaly being detected, anomaly fix being started etc.
+Currently, users are [able to set](https://strimzi.io/docs/operators/latest/full/deploying.html#setting_up_alerts_for_anomaly_detection) the notifier to one of those included with Cruise Control (SelfHealingNotifier, AlertaSelfHealingNotifier, SlackSelfHealingNotifier etc.).
```

Comment:
Wouldn't the information concerning the notifier feature be more relevant in the proposal section instead of the motivation section here where we describe the motivation behind why we want to add self-healing support? Unless we are adding support for Cruise Control notifiers as part of this proposal as well?

Two questions here:

(a) Can we use the Cruise Control notifiers feature on its own in Strimzi today without adding self-healing support?

(b) I understand that we need Cruise Control's notifier feature for supporting the self-healing feature, but wouldn't the sentences about the notifier feature be more relevant in the proposal's implementations section instead of here?